### ISSUE TITLE
Find better name for "testUpdate" and others

### ISSUE THREAD

**Issue by salutis** [salutis/swift-elm-architecture#199]
We currently have `testStart`, `testUpdate` and `testRender`. This is nice but it clashes with XCTest's naming conventions, i.e. each test start with the `test` prefix.

CC: @andris-zalitis

**Comment by salutis** [comment]
Waiting for #201.

API sketch:

```
expect(_ expectation: Expectation, state: State, event: Event?) -> Test<Result, Program>

// where Result == Idle
idle?.state
idle?.actions

// where Result == Loop
loop?.state
loop?.actions

// where Result == Error
stop?.error

let idle = expect(.idle, state: .init(...), event: .didSomething)
expect(idle?.state, equals: 42)
```

**Comment by salutis** [comment]
Fixed for `update` in #202.
